two_factor/forms.py

```diff
@@ -152,6 +160,22 @@ def __init__(self, user, initial_device, **kwargs):
                 label=label
             )
 
+    def clean_device_id(self):
+        if self.data.get("device_id"):
+            try:
+                for user_device in devices_for_user(self.user):
```

Comment:
Given that we know the device ID, couldn't we just fetch it directly rather than iterating over all devices like this?

Reply:
Good point, I have updated the PR to use `Device.from_persistent_id` instead.